### PR TITLE
Bump build target to 2021.1

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -161,45 +163,51 @@ allprojects {
     testAnnotationProcessor(betterStringsCoords)
   }
 
-  gradle.projectsEvaluated {
```
Comment (PawelLipski) [inline build.gradle:164]: This `gradle.projectsEvaluated` didn't seem to have any effect here, probably my mindless copy-paste from stackoverflow

File: build.gradle [hunk 2]
```diff
@@ -161,45 +163,51 @@ allprojects {
...
-        exceptionFormat TestExceptionFormat.FULL
-        showCauses true
-        showExceptions true
-        showStackTraces true
-      }
+  tasks.withType(KotlinCompile) {
+    // Note that unlike `options.release` for `JavaCompile` tasks,
+    // there seems to be no good way to prevent the compiled Kotlin code from using the APIs
+    // of whatever Java version is used to run the compiler
+    // (https://stackoverflow.com/questions/61266285/set-kotlins-java-api-version-with-gradle).
```
Comment (PawelLipski) [inline build.gradle:200]: :this-is-fine:

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/DeleteBranchOnSlideOutSuggestionDialog.kt [hunk 3]
```diff
@@ -18,7 +18,11 @@ data class SlideOutOptions(
 class DeleteBranchOnSlideOutSuggestionDialog(project: Project) :
     DialogWrapper(project, /* canBeParent */ true) {
 
-  private var remember = false
+  // Must NOT be private to prevent:
+  // java.lang.IllegalAccessException: class
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/DeleteBranchOnSlideOutSuggestionDialog.kt:22]: In line 84, no idea why so 😯 as it used to work on kotlin 1.3

File: scripts/prohibit-unused-property-keys [hunk 4]
```diff
@@ -13,7 +13,7 @@ for key in $(
   grep -Po '.*(?==)' -- $bundle \
   | grep -Evx 'action\.GitMachete\.[A-Za-z]+\.(description|GitMachete(ContextMenu|Toolbar).text|text)'
 ); do
-  if ! git grep -qF "$key" -- ":!$bundle"; then
+  if ! git grep -qF "$key" -- '*.java' '*.kt' ":!$bundle"; then
```
Comment (PawelLipski) [inline scripts/prohibit-unused-property-keys:16]: Just to speed up this script a bit (3.0s->1.5s)

File: build.gradle [hunk 5]
```diff
@@ -23,15 +23,15 @@ ext {
...
+    latestMinorsOfOldSupportedMajors: ['2020.1.4', '2020.2.4', '2020.3.3'],
+    latestStable: '2021.1',
     // Note that we have to use a "fixed snapshot" version X.Y.Z-EAP-SNAPSHOT (e.g. 211.4961.33-EAP-SNAPSHOT)
     // rather than a "rolling snapshot" X-EAP-SNAPSHOT (e.g. 211-EAP-SNAPSHOT)
     // to ensure that the builds are reproducible.
     // EAP-CANDIDATE-SNAPSHOTs can be used for binary compatibility checks,
     // but for some reason aren't resolved in UI tests.
     // Generally, see https://www.jetbrains.com/intellij-repository/snapshots/ -> Ctrl+F idea
-    eapOfLatestSupportedMajor: '211.6693.65-EAP-SNAPSHOT'
+    eapOfLatestSupportedMajor: null
```
Comment (mkondratek) [inline build.gradle:34]: ?

**Review by mkondratek — APPROVED** [review]

File: build.gradle (hunk 5 above)
Comment (PawelLipski) [reply build.gradle:34]: We don't have any EAP to test against currently... until a one is released for 2021.2

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/DeleteBranchOnSlideOutSuggestionDialog.kt (hunk 3 above)
Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/DeleteBranchOnSlideOutSuggestionDialog.kt:22]: Whoops there are more places like that... SlideInDialog has the same runtime error... not good...

Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/dialogs/DeleteBranchOnSlideOutSuggestionDialog.kt:22]: Ok, I banned all private fields in Kotlin code xDDDDD
No idea what would be/if there is any better solution ;p

File: build.gradle [hunk 6]
```diff
@@ -537,7 +537,7 @@ dependencies {
 
   testImplementation group: 'com.tngtech.archunit', name: 'archunit', version: archUnitVersion
 
-  // kotlin-reflect is NOT added by Gralde Kotlin plugin 1.4.x automatically,
+  // kotlin-reflect is NOT added by Gradle Kotlin plugin 1.4.x automatically,
```
Comment (mkondratek) [inline build.gradle:540]: `Mieć nóż na Gradle`

File: scripts/prohibit-kotlin-private-fields [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+self_dir=$(cd "$(dirname "$0")" &>/dev/null; pwd -P)
+source "$self_dir"/utils.sh
+
+if git grep 'private va[lr]' -- '*.kt'; then
+  die 'Private fields are apparently unsafe (likely to cause a runtime error) in Kotlin due to https://youtrack.jetbrains.com/issue/KT-29016'
```
Comment (mkondratek) [inline scripts/prohibit-kotlin-private-fields:9]: :open_mouth:

**Review by mkondratek — APPROVED** [review]

File: build.gradle (hunk 6 above)
Comment (PawelLipski) [reply build.gradle:540]: suchar

File: scripts/prohibit-kotlin-private-fields (hunk 7 above)
Comment (PawelLipski) [reply scripts/prohibit-kotlin-private-fields:9]: 💩

**Pull request merged**